s4/clarity/lims.py

```diff
@@ -45,28 +39,35 @@ class LIMS(object):
     :ivar ElementFactory permissions: Factory for :class:`s4.clarity.permission.Permission`
     """
 
-    _HOST_RE = re.compile(r'https?://([^/:]+)')
-
     def __init__(self, root_uri, username, password, dry_run=False, insecure=False, log_requests=False):
-        if root_uri.endswith("/"):
-            self.root_uri = root_uri[:-1]  # strip off /
-        else:
-            self.root_uri = root_uri
+        """
+        Constructs a new LIMS object. This will provide an interface to the Clarity LIMS server, found
+        at the Root URI. The user name and password provided will be used to authenticate with Clarity and
+        all actions taken by the script will be done as that user.
+
+        If a Dry Run is selected then no data will be sent back to Clarity.
+
+        If Insecure is selected then the SSL Certificate used does not need to be signed by a signing authority.
+        This is useful in the case where you are operating against a development server using a self signed cert.
+
+        Log Requests will create log entries for each HTTP request made.
```

Comment:
better style in general here would be to refer to the parameters by their actual names with backticks, like `log_requests`, `insecure`, etc.